sam.c

```diff
@@ -4614,23 +4625,26 @@ uint8_t *bam_aux_get(const bam1_t *b, const char tag[2])
     return NULL;
 }
 
-// s MUST BE returned by bam_aux_get()
 int bam_aux_del(bam1_t *b, uint8_t *s)
 {
-    uint8_t *p, *aux;
-    int l_aux = bam_get_l_aux(b);
-    aux = bam_get_aux(b);
-    p = s - 2;
-    s = skip_aux(s, aux + l_aux);
-    if (s == NULL) goto bad_aux;
-    memmove(p, s, l_aux - (s - aux));
-    b->l_data -= s - p;
-    return 0;
+    s = bam_aux_remove(b, s);
+    return (s || errno == ENOENT)? 0 : -1;
```

Comment:
What if errno is already `ENOENT` on entry?  This is a change of behaviour as it will now fail in situations it previously worked.

I think we should take a temporary copy of errno, set errno to zero and check after `bam_aux_remove`,  with a reset after if appropriate.

Reply:
The post-condition of `bam_aux_remove()` is that either the return value is non-NULL or `errno` will have been set inside `bam_aux_remove()` to indicate why it returned NULL. So there is no change in behaviour as any pre-existing `errno` value is never consulted.

(Both `bam_aux_del()` and `bam_aux_remove()` have a pre-condition that their `s` argument must be non-NULL — and `bam_aux_del()` has always has this requirement. I should probably add a note to the doco for both functions making this explicit.)

Reply:
Sorry that's my silly mental-block or misreading of the OR statement.  You're correct.  Thanks

Reply:
“Non-NULL” note added to both functions.